src/runtime/HalideBuffer.h

```diff
@@ -862,37 +832,24 @@ class Buffer {
         initialize_shape_from_array_shape(buf.dimensions - 1, vals);
     }
 
-    /** Initialize an Buffer of runtime type from a pointer and some
+    /** Initialize a Buffer of runtime type from a pointer and some
      * sizes. Assumes dense row-major packing and a min coordinate of
      * zero. Does not take ownership of the data and does not set the
      * host_dirty flag. */
     template<typename ...Args,
              typename = typename std::enable_if<AllInts<Args...>::value>::type>
-    explicit Buffer(halide_type_t t, add_const_if_T_is_const<void> *data, int first, Args&&... rest) {
-        if (!T_is_void) {
-            assert(static_halide_type() == t);
-        }
-        buf.type = t;
-        buf.dimensions = 1 + (int)(sizeof...(rest));
-        buf.host = (uint8_t *)data;
-        make_shape_storage();
-        initialize_shape(0, first, int(rest)...);
-    }
+    explicit Buffer(halide_type_t t, add_const_if_T_is_const<void> *data, int first, Args&&... rest)
```

Comment:
The constructors that take raw pointers or halide_buffer_t need to compile to very few lines of assembly and do no dynamic allocations, and were carefully designed this way (hence all the variadic junk instead of std::vector). The intended use case is inside a cheap extern stage called per small tile. There are cases of this in the wild.

This is why I was worried about overhead of this path in the other PR.

Reply:
Good to know. I'll withdraw this in favor of one that adds comments to that effect.